src/rgw/rgw_cors_swift.h

```diff
@@ -31,35 +31,38 @@ class RGWCORSConfiguration_SWIFT : public RGWCORSConfiguration
     ~RGWCORSConfiguration_SWIFT() {}
     int create_update(const char *allow_origins, const char *allow_headers, 
                   const char *expose_headers, const char *max_age) {
-      set<string> o, h, oc;
+      set<string> o, h;
       list<string> e;
       unsigned long a = CORS_MAX_AGE_INVALID;
       uint8_t flags = RGW_CORS_ALL;
 
-      string ao = allow_origins;
-      get_str_set(ao, oc);
-      if (oc.empty())
+      auto func = [&o] (auto host) {
+        if (validate_name_string(host) == 0) {
+          o.emplace(host.begin(), host.end());
+        }
+      };
```

Comment:
```suggestion
      int nr_invalid_names = 0;
      auto add_host = [&nr_invalid_names, &o] (auto host) {
        if (validate_name_string(host) == 0) {
          o.emplace(std::string{host});
        } else {
          nr_invalid_names++;
        }
      };
```

Reply:
Fixed. Sorry to break the original logic here.